modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -56,14 +58,16 @@ public class DataStreamIndexSettingsProvider implements IndexSettingProvider {
     }
 
     @Override
-    public Settings getAdditionalIndexSettings(
+    public void onCreateIndex(
```

Comment:
I think I prefer the old name (or something like `getAdditionalIndexMetadata` to indicate it's both settings and metadata), because we call this method in other places, like simulation and validation, not just when an index is created

Reply:
One reason to change it was also that the `get` prefix felt wrong now that the method returns void. I've changed it to `provideAdditionalIndexMetadata` now (after all it's a Provider).